tests/components/bmw_connected_drive/test_sensor.py

```diff
@@ -0,0 +1,47 @@
+"""Test BMW sensors."""
+import pytest
+
+from homeassistant.core import HomeAssistant
+from homeassistant.util.unit_system import IMPERIAL_SYSTEM, METRIC_SYSTEM
+
+from . import setup_mock_component
+
+
+@pytest.mark.parametrize(
+    "entity_id,metric,imperial",
+    [
+        ("sensor.x3_xdrive30e_remaining_range_total", ("516", "km"), ("320.63", "mi")),
```

Comment:
Consider including the unit system as a parameter and extracting the expected values to a single case per parameter case. Ie, double up the lines of entities here and have a single expected value per line and one unit system per line. That's more clear and we then don't need tear down code in the test.